tests/acceptance/error-handling-test.js

```diff
@@ -0,0 +1,90 @@
+    andThen(function() {
+      $('#submit-query').click();
+      assert.equal($('.message').text().toLowerCase().indexOf('does not specify a valid date') > -1, true, "The page issued an error.");
+      visit('/discover/aggregate?obs_date__ge=2016-06-18&obs_date__le=201asgasd12&location_geom__within=%7B%22type%22%3A%22Feature%22%2C%22properties%22%3A%7B%7D%2C%22geometry%22%3A%7B%22type%22%3A%22Polygon%22%2C%22coordinates%22%3A%5B%5B%5B-87.80672550201416%2C41.74262728637672%5D%2C%5B-87.80672550201416%2C41.97582726102573%5D%2C%5B-87.45653629302979%2C41.97582726102573%5D%2C%5B-87.45653629302979%2C41.74262728637672%5D%2C%5B-87.80672550201416%2C41.74262728637672%5D%5D%5D%7D%7D');
+      andThen(function () {
+        assert.equal(currentRouteName(), 'discover.index', 'The page returns to index. (obs_date__le)');
+        visit('/discover?obs_date__ge=2016-06-18&obs_date__le=201asgasd12&location_geom__within=%7B%22type%22%3A%22Feature%22%2C%22properties%22%3A%7B%7D%2C%22geometry%22%3A%7B%22type%22%3A%22Polygon%22%2C%22coordinates%22%3A%5B%5B%5B-87.80672550201416%2C41.74262728637672%5D%2C%5B-87.80672550201416%2C41.97582726102573%5D%2C%5B-87.45653629302979%2C41.97582726102573%5D%2C%5B-87.45653629302979%2C41.74262728637672%5D%2C%5B-87.80672550201416%2C41.74262728637672%5D%5D%5D%7D%7D');
+        andThen(function () {
+          $('#submit-query').click();
+          assert.equal($('.message').text().toLowerCase().indexOf('does not specify a valid date') > -1, true, "The page issued an error.");
+        });
+      });
+    });
+  });
+});
+
+test('Attempting to query with invalid JSON returns to index.', function(assert){
+  visit('/discover/aggregate?location_geom__within=%7B%22type%22%3A%22Feature%22%3A%7B%7D%2C%22geometry%22%3A%7B%22type%22%3A%22Polygon%22%2C%22coordinates%22%3A%5B%5B%5B-87.80672550201416%2C41.74262728637672%5D%2C%5B-87.80672550201416%2C41.97582726102573%5D%2C%5B-87.45653629302979%2C41.97582726102573%5D%2C%5B-87.45653629302979%2C41.74262728637672%5D%2C%5B-87.80672550201416%2C41.74262728637672%5D%5D%5D%7D%7D');
+  andThen(function() {
+    assert.equal(currentRouteName(), 'discover.index', 'The page returns to index.');
+    //TODO: Add an assertion here that checks that a error was issued.
+  });
+});
+
+test('Attempting to query without a geoJSON returns to index and issues an error.', function(assert){
+  visit('/discover/aggregate');
+  andThen(function() {
+    assert.equal(currentRouteName(), 'discover.index', 'The page returns to index.');
+    visit('/discover');
+    $('#submit-query').click();
+    assert.equal($('.message').text().toLowerCase().indexOf('draw a shape on the map') > -1, true, "The page issued an error.");
```

Comment:
Don't like that we're testing the content of the error message. What happens when we change the message? Maybe better just to test that there is a non-empty message.